Realm/RLMResults.mm

```diff
@@ -282,6 +289,28 @@ - (NSUInteger)indexOfObject:(RLMObject *)object {
     });
 }
 
+- (id)valueForKeyPath:(NSString *)keyPath {
+    if ([keyPath characterAtIndex:0] == '@') {
+        NSRange operatorRange = [keyPath rangeOfString:@"." options:NSLiteralSearch];
+        if (operatorRange.location == NSNotFound) {
+            if ([keyPath isEqualToString:@"@count"]) {
+                return @(self.count);
+            }
+            else {
+                NSAssert(false, @"Unsupported KVC collection operator found in key path '%@'", keyPath);
```

Comment:
Would you do that as well for the remaining `NSAsserts` with actual conditions? Is there a point in being able to catch those?